package mcmc;

message VectorStorage {
  required bytes storage = 1;
}

message RpmProperties {
  required uint32 rows = 1;
  required uint32 cols = 2;
  required uint32 rows_in_block = 3;
}

message OpenClRandomStorage {
  required VectorStorage seeds = 1;
}

message BetaProperties {
  required uint32 count_calls = 1;
  required double theta_sum_time = 2;
  required double grads_partial_time = 3;
  required double grads_sum_time = 4;
  required double update_theta_time = 5;
  required double normalize_time = 6;
}

message PhiProperties {
  required uint32 count_calls = 1;
  required double update_phi_time = 2;
  required double update_pi_time = 3;
}

message PerplexityProperties {
  required uint32 count_calls = 1;
  required double ppx_time = 2;
  required double accumulate_time = 3;
}

message SampleStorage {
  required bytes edges = 1;
  required bytes nodes_vec = 2;
  required uint32 seed = 3;
}

message LearnerProperties {
  required uint32 stepCount = 1;
  required uint64 time = 2;
  required uint64 samplingTime = 3;
  required int32 phase = 4;
  required double weight = 5;
}
